### PR TITLE
make API debuggable

### PR DESCRIPTION
**Pull request by zdevito**

### CODE REVIEW

**Review by prigoyal — APPROVED** [review]

**Comment by prigoyal** [comment]
looks good and cleaner to me! thanks for verbosity!

File: torch/csrc/jit/python_tracer.cpp [hunk 1]
```diff
@@ -79,11 +79,14 @@ void initPythonTracerBindings(PyObject* module_) {
       ss << *s.graph;
       return ss.str();
     })
-    .def("export", [](TracingState& s) {
-      return py::bytes(ExportGraph(s.graph, s.buffer_map, {}));
+    .def("export", [](TracingState& s, bool verbose) {
```
Comment (ezyang) [inline torch/csrc/jit/python_tracer.cpp:82]: I guess we could probably fix the CI failure by making this optional parameter. Another way is to update all the trace.export() invocations to call the higher level API.

File: torch/csrc/toffee/export.cpp [hunk 2]
```diff
@@ -298,8 +301,12 @@ static void encodeGraph(toffee::GraphProto * p_g, std::shared_ptr<Graph> & g, co
 // Exports a graph to ToffeeIR
 std::string ExportGraph(std::shared_ptr<Graph>& g_,
                         const std::unordered_map<void*, Node*>& buffer_map,
-                        const std::vector<at::Tensor> & initializers) {
+                        const std::vector<at::Tensor> & initializers,
+                        bool verbose) {
   auto g = ToToffeeIR(g_, buffer_map);
+  if(verbose) {
+    std::cout << *g << "\n";
```
Comment (ezyang) [inline torch/csrc/toffee/export.cpp:308]: Should verbose also print out `g_`?

**Comment by ezyang** [comment]
LGTM but THE CI MUST BE APPEASED

File: torch/toffee.py [hunk 3]
```diff
@@ -46,8 +40,8 @@ def _export(model, args, f, export_params=True, kwargs=None):
     trace, torch_out = torch.jit.record_trace(model, *args, **kwargs)
     # TODO: Don't allocate a in-memory string for the protobuf
     if export_params:
-        proto = trace.export(model.state_dict().values())
+        proto = trace.export(model.state_dict().values(), verbose)
     else:
-        proto = trace.export()
+        proto = trace.export(verbose)
     torch.serialization._with_file_like(f, "wb", lambda f: f.write(proto))
     return torch_out
```
Comment (apaszke) [inline torch/toffee.py:47]: Not returning the output was a conscious choice for this API. Please keep it as `_export` for tests

Comment (zdevito) [reply torch/toffee.py:47]: This conscious choice was done and accepted without review by either Priya, Lu or myself who have been using these APIs over the last week. How can we debug a model if we can't see if an output matches? How can anyone else?

Comment (apaszke) [reply torch/toffee.py:47]: IIRC Ed has communicated that to both Priya and Lu in the VC. I can't see why retrieving model inputs would be any harder than `model(inputs)`

**Pull request merged**